readthedocs/storage/s3_storage.py

```diff
@@ -19,23 +19,6 @@
 from .mixins import OverrideHostnameMixin, S3PrivateBucketMixin
 
 
-class S3BuildMediaStorage(BuildMediaStorageMixin, OverrideHostnameMixin, S3Boto3Storage):
```

Comment:
It seems that you are deleting the wrong backend here. This one is for `BuildMedia`, and we need to modify the one for static files.